src/System.Device.Gpio/System/Device/Gpio/AlternatePinMode.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class AlternatePinMode : IEquatable<AlternatePinMode>
+    {
+        /// <summary>
+        /// Default Gpio mode
+        /// </summary>
+        public static AlternatePinMode Gpio = new AlternatePinMode("GPIO", 0);
```

Comment:
since you're using statics this is fine as you don't need to allocate it all the time. I'll just double check this is immutable

Reply:
Yes, an instance of AlternatePinMode is immutable.